.github/workflows/build.yml

```diff
@@ -55,57 +62,146 @@ jobs:
       - name: Build docker image
         run: | 
           SAMPLE_LIST=(hello-kubernetes distributed-calculator pub-sub bindings observability secretstore)
-
           for sample in "${SAMPLE_LIST[@]}"; do
             echo "Building image for ${sample}..."
             pushd ${sample}
             make build
             popd
           done
-      - name: docker login
-        run: |
-          docker login -u ${{ secrets.DOCKER_REGISTRY_ID }} -p ${{ secrets.DOCKER_REGISTRY_PASS }}
-      - name: publish image to dockerhub
+     - name: GitHub container registry login
+        if: matrix.target_os != 'darwin'
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push docker images to GitHub container registry
+        if: matrix.target_os != 'darwin'
         run: |
-          SAMPLE_LIST=(hello-kubernetes distributed-calculator pub-sub bindings observability secretstore)
-
-          for sample in "${SAMPLE_LIST[@]}"; do
-            echo "Push image for ${sample}..."
-            pushd ${sample}
-            make push
-            popd
-          done
+          echo "Build docker image and push image..."
+          make docker-push TARGET_OS=${{ matrix.target_os }} TARGET_ARCH=${{ matrix.target_arch }} DAPR_REGISTRY=ghcr.io/${{ github.repository_owner }} DAPR_TAG=${{ env.REL_VERSION }}
```

Comment:
We can check this file for what make command we already have for pushing images.
https://github.com/dapr/quickstarts/blob/870ed621b843708cab9a78f7ec4cd122575957c2/docker.mk#L1
`make docker-push`  is not here I guess. We can think of adding ` docker-push` or reuse what we have already. 
Overall flow would look like this - 

1. Build docker images - Already being done
2. Docker login - Already being done
3. ghcr login - It is added in this PR
4. push images to docker - Already being done
5. push images to ghcr - We only need to add this.

We can combine 4 & 5 together or keep it separate.